### PR TITLE
Fix avatar URLs

### LINKED ISSUE #3068: Misformatted gravatar URLs
- Gitea version (or commit ref): master
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant

## Description

If gravatar is enabled, but no federated avatar service is used, then the avatar URLs for organizations sometime have misformatted query parameters. This is because we naively append `?s=(size)` to the end of avatar URLs, which may already query parameters. I will have a fix soon.

## Screenshots

bad_avatar
bad_avatar_url

### PR DESCRIPTION
**Pull request by ethantkoenig**
Fixes #3068. This PR also
* Does some refactoring to use the `net/url` library instead of manually assembling URLs.
* Uses identicons for the default avatar when a federated avatar service is enabled. Previously, the default was identicons if a federated service was *not* enabled, but the Gravatar logo if a federated service was enabled.

### CODE REVIEW

File: modules/base/tool.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ import (
 	"github.com/Unknwon/com"
 	"github.com/Unknwon/i18n"
 	"github.com/gogits/chardet"
+	"net/url"
```
Comment (lunny) [inline modules/base/tool.go]: import order

File: modules/base/tool_test.go [hunk 2]
```diff
@@ -9,6 +9,7 @@ import (
 	"github.com/Unknwon/i18n"
 	macaroni18n "github.com/go-macaron/i18n"
 	"github.com/stretchr/testify/assert"
+	"net/url"
```
Comment (lunny) [inline modules/base/tool_test.go]: import order

File: modules/base/tool.go (hunk 1 above)
Comment (ethantkoenig) [reply modules/base/tool.go]: done

File: modules/base/tool_test.go (hunk 2 above)
Comment (ethantkoenig) [reply modules/base/tool_test.go]: done

**Comment by lunny** [comment]
LGTM

**Comment by lafriks** [comment]
LGTM

**Comment by lunny** [comment]
also fixed #2941

**Comment by ufdada** [comment]
Any chance to get a backport to 3.1.x?

**Comment by lafriks** [comment]
@ethantkoenig can you backport?

**Pull request merged**